### PR TITLE
rose macro: added suite-only option

### LINKED ISSUE #1953: rose macro: option to run on the suite only and not its apps
#1884 added the ability to run `rose macro` over an entire suite (rose-suite.conf included) not just an individual app.

If a user wants to run `rose macro` on the rose-suite.conf they must also run it on all apps which is a tad inconvenient.

### PR DESCRIPTION
**Pull request by oliver-sanders**
Closes #1953 

Added `--suite-only` option to rose macro for running suite level macros only.

@arjclark Please review
@matthewrmshin Please review

### CODE REVIEW

File: lib/python/rose/macro.py [hunk 1]
```diff
@@ -1335,7 +1335,7 @@ def _report_error(exception=None, text=""):
     )
 
 
-def scan_rose_directory(conf_dir):
+def scan_rose_directory(conf_dir, top_level_only=False):
```
Comment (matthewrmshin) [inline lib/python/rose/macro.py]: Should `top_level_only` be `suite_only`? (Or vice versa.) Using the same variable name makes it easier to search for the functionality in the future.

File: t/rose-macro/00-null.t [hunk 2]
```diff
@@ -221,6 +221,19 @@ file_cmp "$TEST_KEY.err" "$TEST_KEY.err" <<'__ERR__'
     env=SUITE=sweet
         Not an integer: 'sweet'
 __ERR__
+
+# run with --suite-only option
+TEST_KEY=$TEST_KEY_BASE-suite-only
+run_pass "$TEST_KEY" rose macro -C $TEST_DIR/$TEST_SUITE/app --suite-only
+cat $TEST_KEY.out
```
Comment (matthewrmshin) [inline t/rose-macro/00-null.t]: Debug statement can be removed?

**Comment by oliver-sanders** [comment]
Closes #1570

**Comment by arjclark** [comment]
Looks ok to me. Just needs documenting.

**Comment by arjclark** [comment]
Looks OK to me, documentation added. No problems from the test-battery in my environment.

**Pull request merged**